src/Test.FunctionalTests.Sarif/TestData/Multitool/ValidateCommand/Inputs/SARIF1008.PhysicalLocationPropertiesMustBeConsistent_Invalid.sarif

```diff
@@ -28,6 +28,154 @@
+                  "endLine": 10
+                },
+                "contextRegion": {
+                  "startLine": 2,
+                  "endLine": 4
+                }
+              }
+            }
+          ]
+        },
+        {
+          "ruleId": "TST0004",
+          "level": "error",
+          "message": {
+            "text": "Bad result - fully overlapping region and contextregion (line/column based)."
+          },
+          "locations": [
+            {
+              "physicalLocation": {
+                "artifactLocation": {
+                  "uri": "src/test.c"
+                },
+                "region": {
+                  "startLine": 2,
+                  "endLine": 4
+                },
+                "contextRegion": {
+                  "startLine": 2,
+                  "endLine": 4
+                }
+              }
+            }
+          ]
+        },
+        {
+          "ruleId": "TST0005",
+          "level": "error",
+          "message": {
+            "text": "Bad result - text contextregion smaller than region (offset based)."
+          },
+          "locations": [
+            {
+              "physicalLocation": {
+                "artifactLocation": {
+                  "uri": "src/test.c"
+                },
+                "region": {
+                  "charOffset": 15,
+                  "charLength": 30
+                },
+                "contextRegion": {
+                  "charOffset": 15,
+                  "charLength": 20
+                }
+              }
+            }
+          ]
+        },
+        {
+          "ruleId": "TST0006",
+          "level": "error",
+          "message": {
+            "text": "Bad result - non overlapping text region and contextregion (offset based)."
+          },
+          "locations": [
+            {
+              "physicalLocation": {
+                "artifactLocation": {
+                  "uri": "src/test.c"
+                },
+                "region": {
+                  "charOffset": 1,
+                  "charLength": 10
+                },
+                "contextRegion": {
+                  "charOffset": 15,
+                  "charLength": 20
+                }
+              }
+            }
+          ]
+        },
+        {
+          "ruleId": "TST0007",
+          "level": "error",
+          "message": {
+            "text": "Bad result - fully overlapping binary region and contextregion (offset based)."
+          },
+          "locations": [
+            {
+              "physicalLocation": {
+                "artifactLocation": {
+                  "uri": "src/test.c"
+                },
+                "region": {
+                  "byteOffset": 10,
+                  "byteLength": 100
+                },
+                "contextRegion": {
+                  "byteOffset": 10,
+                  "byteLength": 100
+                }
+              }
+            }
+          ]
```

Comment:
>] [](start = 10, length = 1)

Let's make the tests more consistent. For each of the three region kinds (line/column, text offset/length, binary offset/length) let's have three invalid cases:
1. Region and context region are identical.
2. Contenxt region partially overlaps region (context region starts after and ends region, or context region starts before and ends before region, I don't care which).
3. Context region is smaller than region (context region starts after and ends before region).

Reply:
noted. updating now

---
In reply to: [443870279](https://github.com/microsoft/sarif-sdk/pull/1925#discussion_r443870279) [](ancestors = 443870279)